### PR TITLE
flowveldepth and nudge value store as netcdf for every 1 hr

### PR DESCRIPTION
**Pull request by AminTorabi-NOAA**
The code store flow velocity and depth, along with nudge values, as NetCDF files every hour. The Flowveldepth DataFrame is updated with nudge values corresponding to each gauge ID, and a NetCDF file is saved for each hour. The name of the file indicates the timestamp.

## Additions

-

## Removals

-

## Changes

-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/troute-nwm/src/nwm_routing/output.py [hunk 1]
```diff
@@ -194,7 +194,10 @@ def nwm_output_generator(
 
     if test:
         flowveldepth.to_pickle(Path(test))
-    
+        
+        nudge = results[0][8]
+        usgs_positions_id = results[0][3][0]
```
Comment (shorvath-noaa) [inline src/troute-nwm/src/nwm_routing/output.py:199]: `results` is a list, I believe you need to loop through it to obtain all the nudge values, no?

File: src/troute-nwm/src/nwm_routing/output.py [hunk 2]
```diff
@@ -194,7 +194,10 @@ def nwm_output_generator(
...
+        nhd_io.write_flowveldepth_netcdf('../LowerColorado_TX/output', flowveldepth, nudge, usgs_positions_id, t0)
```
Comment (shorvath-noaa) [inline src/troute-nwm/src/nwm_routing/output.py:200]: it'd be good to make the `filepath` here a user specified input parameter in the configuration file.

File: src/troute-nwm/src/nwm_routing/output.py [hunk 3]
```diff
@@ -194,10 +195,11 @@ def nwm_output_generator(
 
     if test:
         flowveldepth.to_pickle(Path(test))
+        import pdb; pdb.set_trace()
```
Comment (AminTorabi-NOAA) [inline src/troute-nwm/src/nwm_routing/output.py:198]: Forgot to delete this line. I'll delete it

File: src/troute-nwm/src/nwm_routing/output.py [hunk 4]
```diff
@@ -194,7 +195,11 @@ def nwm_output_generator(
 
     if test:
         flowveldepth.to_pickle(Path(test))
-    
+        import pdb; pdb.set_trace()
+        nudge = np.concatenate([r[8] for r in results])
+        usgs_positions_id = np.concatenate([r[3][0] for r in results])
+        
+        nhd_io.write_flowveldepth_netcdf(Path(qvd_ndg), flowveldepth, nudge, usgs_positions_id, t0)
```
Comment (shorvath-noaa) [inline src/troute-nwm/src/nwm_routing/output.py:202]: Let's move this into it's own logic statement rather than being nested under if test: ...

File: src/troute-nwm/src/nwm_routing/output.py (hunk 3 above)
Comment (shorvath-noaa) [reply src/troute-nwm/src/nwm_routing/output.py:198]: Looks like it's still there?

File: test/LowerColorado_TX/test_AnA.yaml [hunk 5]
```diff
@@ -35,10 +35,10 @@ compute_parameters:
     restart_parameters:
         #----------
         start_datetime: 2021-08-23_13:00
-        wrf_hydro_channel_restart_file        : restart/HYDRO_RST.2021-08-23_12:00_DOMAIN1
+        # wrf_hydro_channel_restart_file        : restart/HYDRO_RST.2021-08-23_12:00_DOMAIN1
         #lite_channel_restart_file            : restart/RESTART.2020082600_DOMAIN1
         wrf_hydro_channel_ID_crosswalk_file   : domain/RouteLink.nc
-        wrf_hydro_waterbody_restart_file      : restart/HYDRO_RST.2021-08-23_12:00_DOMAIN1
+        # wrf_hydro_waterbody_restart_file      : restart/HYDRO_RST.2021-08-23_12:00_DOMAIN1
```
Comment (shorvath-noaa) [inline test/LowerColorado_TX/test_AnA.yaml:41]: can you undo these changes? It's good to leave this test_AnA.yaml file as is unless necessary.

File: test/LowerColorado_TX/test_AnA.yaml [hunk 6]
```diff
@@ -77,8 +77,8 @@ compute_parameters:
             lastobs_output_folder         : lastobs/
         reservoir_da:
             #----------
-            reservoir_persistence_usgs  : True
-            reservoir_persistence_usace : True
+            reservoir_persistence_usgs  : False
+            reservoir_persistence_usace : False
```
Comment (shorvath-noaa) [inline test/LowerColorado_TX/test_AnA.yaml:81]: Same as above, can you undo these changes?

File: src/troute-network/troute/nhd_io.py [hunk 7]
```diff
@@ -1942,3 +1942,89 @@ def write_waterbody_netcdf(
...
+
+                # =========== q,v,d,ndg VARIABLES ===============
+                for counter, var in enumerate(['flowrate', 'velocity', 'depth', 'nudge']):
+                    QVD = ncfile.createVariable(
+                        varname=var,
+                        datatype=np.float32,
+                        dimensions=('feature_id', 'time_step'),
+                    )
+
+                    QVD.units = 'm3/s m/s m m'
```
Comment (shorvath-noaa) [inline src/troute-network/troute/nhd_io.py:2021]: is the unit of `nudge` listed as meters? Doesn't seem right.

File: src/troute-network/troute/nhd_io.py [hunk 8]
```diff
@@ -1942,3 +1942,89 @@ def write_waterbody_netcdf(
...
+       
+        if qvd_filepath:
+            # Open netCDF4 Dataset in write mode
+            with netCDF4.Dataset(
+                filename=f"{qvd_filepath}/{file_name}",
+                mode='w',
+                format='NETCDF4'
+            ) as ncfile:
+
+                # ============ DIMENSIONS ===================
```
Comment (shorvath-noaa) [inline src/troute-network/troute/nhd_io.py:2007]: I'm wondering if we should add some metadata to these files? For instance some information about what exactly time step is (like seconds from a common datetime?). Can you take a look at some of the other functions that create netcdf files to get an idea of metadata options?

**Comment by shorvath-noaa** [comment]
@AminTorabi-NOAA , it looks like you've included changes to CHRTOUT, LAKEOUT, and your new flowveldepth files in this PR. I think it's better to omit these and only include changes to the actually t-route code. Otherwise, when people clone t-route and run this test case, they won't see any changes and so may believe the test didn't run successfully. Can you remove all these changes from this PR?
